### PR TITLE
Hide baselayer module

### PR DESCRIPTION
**Pull request by vllorca**

### CODE REVIEW

File: src/modules/RA/ra-modules.js [hunk 1]
```diff
@@ -141,6 +141,7 @@ export const config = {
     {
       label: 'baseLayer.project',
       requiredModule: 'BaseLayer',
+      // requiredPermissions: 'can_manage_baselayers',
```
Comment (CFarcy) [inline src/modules/RA/ra-modules.js:144]: Commented code should not be present, unless for very precise reasons

File: src/modules/RA/ra-modules.js [hunk 2]
```diff
@@ -99,8 +99,8 @@ export const resources = [
   {
     name: RES_BASELAYER,
     moduleName: 'BaseLayer',
-    ...baseLayerViews,
     endpoint: 'baselayer',
+    ...baseLayerViews,
```
Comment (CFarcy) [inline src/modules/RA/ra-modules.js:103]: What is the reason behind this change?

File: src/modules/RA/ra-modules.js (hunk 1 above)
Comment (vllorca) [reply src/modules/RA/ra-modules.js:144]: I kept this comment, as permissions could be a solution for the ticket and should be discussed with Pauline or Jeremie

Comment (vllorca) [reply src/modules/RA/ra-modules.js:144]: Do you want me to erase this comment ?

File: src/modules/RA/ra-modules.js (hunk 2 above)
Comment (vllorca) [reply src/modules/RA/ra-modules.js:103]: I did this for the object to match the the ones in the Array
It does not change the behavior, do you want me to revert this change ?

File: src/modules/RA/ra-modules.js [hunk 3]
```diff
@@ -150,6 +151,11 @@ export const config = {
   ],
 };
 
+if (process.env.REACT_APP_NOT_DISPLAY_BASELAYERS) {
+  const { menu } = config;
+  config.menu = menu.filter(obj => obj.label !== 'baseLayer.project');
+}
+
```
Comment (dtrucs) [inline src/modules/RA/ra-modules.js:158]: Maybe you could set a more generic stuff by setting a "pseudo array".
For example in the .env file you've got the following line ⬇️ 
`REACT_APP_NOT_DISPLAY_MODULES="baseLayer foo bar"`

And then in the code you can iterate on it
```js
if (process.env.REACT_APP_NOT_DISPLAY_MODULES) {
  const { menu } = config;
  process.env.REACT_APP_NOT_DISPLAY_MODULES.split(' ').forEach(module => {
   config.menu = menu.filter(({ label }) => label !== module);
  }
}
```

Comment (dtrucs) [reply src/modules/RA/ra-modules.js:158]: ok i miss `.project` but it's possible with a .map after the split.
What do you think?

File: src/modules/RA/ra-modules.js (hunk 2 above)
Comment (CFarcy) [reply src/modules/RA/ra-modules.js:103]: Not  necessarily revert the change, but as it is a stylistic change, would be better in a separate commit in my opinion.

File: src/modules/RA/ra-modules.js (hunk 1 above)
Comment (CFarcy) [reply src/modules/RA/ra-modules.js:144]: Even though some change could appear in the work of this PR, commented code should not be present.

File: src/modules/RA/ra-modules.js (hunk 3 above)
Comment (CFarcy) [reply src/modules/RA/ra-modules.js:158]: I like this idea

Comment (vllorca) [reply src/modules/RA/ra-modules.js:157]: I applied the changes
There is now a env variable looking like REACT_APP_NOT_DISPLAY_MODULES="baseLayer.project foo.project bar.project"

Comment (dtrucs) [reply src/modules/RA/ra-modules.js:157]: Yes top!
Maybe you can avoid the repetition of '.project' in the env file and match the correspondance here
```js
const notDisplayModulesProject = !process.env.REACT_APP_NOT_DISPLAY_MODULES.split(' ').map(module => `${module}.project`}
config.menu = menu.filter(({ label }) => notDisplayModulesProject.includes(label));`
```

or but maybe less accurate
```js
config.menu = menu.filter(({ label }) => !process.env.REACT_APP_NOT_DISPLAY_MODULES.split(' ').includes(label.replace('.project', ''));
```

**Comment by vllorca** [comment]
This issue was solved in django-terra-account using a django manager to filter modules and permissions with a disabled_module setting
